core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/rds/RdsUpgradeService.java

```diff
@@ -83,12 +86,16 @@ private RdsUpgradeV4Response alreadyOnLatestAnswer(TargetMajorVersion targetMajo
                 getMessage(CLUSTER_RDS_UPGRADE_ALREADY_UPGRADED, List.of(targetMajorVersion.getMajorVersion())), targetMajorVersion);
     }
 
-    private RdsUpgradeV4Response checkStackStatusAndTrigger(StackView stack, TargetMajorVersion targetMajorVersion) {
+    private RdsUpgradeV4Response validateAndTrigger(StackView stack, TargetMajorVersion targetMajorVersion, String accountId) {
         if (!stack.getStatus().isAvailable() && Status.EXTERNAL_DATABASE_UPGRADE_FAILED != stack.getStatus()) {
             LOGGER.info("Stack {} is not available for RDS upgrade", stack.getName());
             return new RdsUpgradeV4Response(RdsUpgradeResponseType.ERROR, FlowIdentifier.notTriggered(),
                     getMessage(CLUSTER_RDS_UPGRADE_NOT_AVAILABLE), targetMajorVersion);
         }
+        Optional<String> runtimeValidationError = databaseUpgradeRuntimeValidator.validateRuntimeVersionForUpgrade(stack.getStackVersion(), accountId);
```

Comment:
I would prefer if this validation would come before stack status validation. 
Otherwise it could happen that first we only present stack not available but later we present this runtime validation error, which is not a good user experience.